### DISCUSSION TITLE
Adding a mesh to a loaded scene

### DISCUSSION THREAD

**Discussion by matthiasschwartz** [mitsuba-renderer/mitsuba3#380] [Q&A] [answered]
Hello, i'm trying to add a mesh, which i create in the python script, to a scene which i load from an .xml file. 
I tried adding the mesh via the SceneParameters, aswell as scene.shapes(), but it did not work. mi.xml_to_props() also did not work, as it just returned a List, which i could not turn into a scene.
Is this possible in some other way?

**Comment by njroussel** [comment, marked as answer]
Hi

Currently, it is not possible to add a `Mesh` to a `Scene` once it has been loaded.

One nice workaround is, if you have your scene description in a Python `dict`,  you can dynamically add your loaded mesh to the scene description and then load the scene. You can find an example [here](https://mitsuba.readthedocs.io/en/latest/src/how_to_guides/mesh_io_and_manipulation.html#Procedural-mesh) (last cell of the "Procedural mesh" section).

If you're dealing with a XML scene description, you can save your mesh to a PLY using `Mesh.write_to_ply` and then either add it to your XML file or you can read the contents of the XML file and edit the string to add your mesh and then use `mitsuba.load_string`.

**Reply by matthiasschwartz** [reply]
Okay, thanks for the answer! I will do it like that then
